tests/network/bindingplugin_macvtap.go

```diff
@@ -0,0 +1,260 @@
+	})
+
+	BeforeEach(func() {
+		const macvtapBindingName = "macvtap"
+		err := libkvconfig.WithNetBindingPlugin(macvtapBindingName, v1.InterfaceBindingPlugin{
+			DomainAttachmentType: v1.Tap,
+		})
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	BeforeEach(func() {
+		ns := testsuite.GetTestNamespace(nil)
+		Expect(libnet.CreateMacvtapNetworkAttachmentDefinition(ns, macvtapNetworkName, macvtapLowerDevice)).To(Succeed(),
+			"A macvtap network named %s should be provisioned", macvtapNetworkName)
+	})
+
+	It("should successfully create a VM with macvtap interface with custom MAC address", func() {
+		const mac = "02:00:00:00:00:02"
+		vmi, err := createAlpineVMIRandomNode(macvtapNetworkName, mac)
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(vmi.Status.Interfaces).To(HaveLen(1), "should have a single interface")
+		Expect(vmi.Status.Interfaces[0].MAC).To(Equal(mac), "the expected MAC address should be set in the VMI")
+	})
+
+	It("two VMs with macvtap interface should be able to communicate over macvtap network", func() {
+		nodeList := libnode.GetAllSchedulableNodes(kubevirt.Client())
+		Expect(nodeList.Items).NotTo(BeEmpty(), "schedulable kubernetes nodes must be present")
+		nodeName := nodeList.Items[0].Name
+
+		chosenMACHW, err := GenerateRandomMac()
+		Expect(err).ToNot(HaveOccurred())
+		chosenMAC := chosenMACHW.String()
+
+		serverCIDR := "192.0.2.102/24"
+		serverIP, err := libnet.CidrToIP(serverCIDR)
+		Expect(err).ToNot(HaveOccurred())
+
+		_ = createAlpineVMIStaticIPOnNode(nodeName, macvtapNetworkName, "eth0", serverCIDR, &chosenMAC)
+		clientVMI := createAlpineVMIStaticIPOnNode(nodeName, macvtapNetworkName, "eth0", "192.0.2.101/24", nil)
+
+		Expect(libnet.PingFromVMConsole(clientVMI, serverIP)).To(Succeed())
+	})
+
+	Context("VMI migration", func() {
+		var clientVMI *v1.VirtualMachineInstance
+
+		BeforeEach(func() {
+			checks.SkipIfMigrationIsNotPossible()
+		})
+
+		BeforeEach(func() {
+			macAddressHW, err := GenerateRandomMac()
+			Expect(err).ToNot(HaveOccurred())
+			macAddress := macAddressHW.String()
+			clientVMI, err = createAlpineVMIRandomNode(macvtapNetworkName, macAddress)
+			Expect(err).NotTo(HaveOccurred(), "must succeed creating a VMI on a random node")
+		})
+
+		It("should be successful when the VMI MAC address is defined in its spec", func() {
+			By("starting the migration")
+			migration := libmigration.New(clientVMI.Name, clientVMI.Namespace)
+			migration = libmigration.RunMigrationAndExpectToCompleteWithDefaultTimeout(kubevirt.Client(), migration)
+
+			// check VMI, confirm migration state
+			libmigration.ConfirmVMIPostMigration(kubevirt.Client(), clientVMI, migration)
+		})
+
+		Context("with live traffic", func() {
+			var serverVMI *v1.VirtualMachineInstance
+			var serverVMIPodName string
+			var serverIP string
+
+			macvtapIfaceIPReportTimeout := 4 * time.Minute
+
+			waitVMMacvtapIfaceIPReport := func(vmi *v1.VirtualMachineInstance, macAddress string, timeout time.Duration) (string, error) {
```

Comment:
What is the reason to use a lambda?:)

Reply:
Haha :sweat_smile: , I moved the test and all the content as is.